### PR TITLE
Implemented Frame.waitForSelector tests

### LINKED ISSUE #140: Implement Frame.waitForSelector
https://github.com/GoogleChrome/puppeteer/blob/v1.0.0/test/test.js#L717

### PR DESCRIPTION
**Pull request by Meir017**
closes #140

### CODE REVIEW

File: lib/PuppeteerSharp.Tests/Frame/WaitForSelectorTests.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            var watchdog = frame2.WaitForSelectorAsync("div").ContinueWith(_ => added = true);
+            Assert.False(added);
+
+            await frame1.EvaluateFunctionAsync(AddElement, "div");
+            Assert.False(added);
+
+            await frame2.EvaluateFunctionAsync(AddElement, "div");
+            Assert.True(added);
+
+            await watchdog;
```
Comment (kblok) [inline lib/PuppeteerSharp.Tests/Frame/WaitForSelectorTests.cs]: Original test doesn't have this await

File: lib/PuppeteerSharp/Page.cs [hunk 2]
```diff
@@ -160,6 +160,12 @@ private Page(Session client, Target target, FrameTree frameTree, bool ignoreHTTP
             return await context.EvaluateFunctionHandleAsync(pageFunction, args);
         }
 
+        public async Task EvaluateOnNewDocumentAsync(string pageFunction, params object[] args)
+        {
+            var source = Helper.EvaluationString(pageFunction, args);
```
Comment (kblok) [inline lib/PuppeteerSharp/Page.cs]: I think `EvaluationString` in Puppeteer tries to serialize a javascript function, I don't know if we need to do that.

File: lib/PuppeteerSharp.Tests/Frame/WaitForSelectorTests.cs [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+        {
+            await Page.EvaluateOnNewDocumentAsync(@"function() {
+                document.querySelector = null;
+            }");
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            var exception = await Assert.ThrowsAnyAsync<PuppeteerException>(() => Page.WaitForSelectorAsync("*"));
+            Assert.Contains("document.querySelector is not a function", exception.Message);
+        }
+
+        [Fact(Skip = "FrameUtils.DetachFrame hangs :(")]
```
Comment (kblok) [inline lib/PuppeteerSharp.Tests/Frame/WaitForSelectorTests.cs]: Do you need a hand here?

Comment (Meir017) [reply lib/PuppeteerSharp.Tests/Frame/WaitForSelectorTests.cs]: Not sure yet. I didn't give a lot of time. I just figured out what the problem

**Comment by kblok** [comment]
This will turn this PR green https://github.com/kblok/puppeteer-sharp/commit/0faec394f762942a77fc8e2850933a50505e18cc (I Hope 😂 )

**Pull request merged**